rocksdb/options/readopts.nim

```diff
@@ -0,0 +1,41 @@
+# Nim-RocksDB
+# Copyright 2024 Status Research & Development GmbH
+# Licensed under either of
+#
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+#  * GPL license, version 2.0, ([LICENSE-GPLv2](LICENSE-GPLv2) or https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html)
+#
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+{.push raises: [].}
+
+import
+  ../lib/librocksdb
+
+type
+  ReadOptionsPtr* = ptr rocksdb_readoptions_t
+
+  ReadOptionsRef* = ref object
+    cPtr: ReadOptionsPtr
+
+proc newReadOptions*(): ReadOptionsRef =
+  ReadOptionsRef(cPtr: rocksdb_readoptions_create())
+
+template isClosed*(readOpts: ReadOptionsRef): bool =
+  readOpts.cPtr.isNil()
+
+proc cPtr*(readOpts: ReadOptionsRef): ReadOptionsPtr =
+  doAssert not readOpts.isClosed()
+  readOpts.cPtr
+
+# TODO: Add setters and getters for read options properties.
```

Comment:
RocksDb is very configurable and so each of these various options types can be updated in the future to support setting the required fields.